Hca leading zero fix dateinput

akainic opened the pull request:
During usability testing users were confused by the input fields for month and day not accepting leading zeros. The options were either to change the input type to a text field or to use dropdowns. As the original form uses dropdowns for both month and day, I decided to follow that format. I also created an `ErrorableNumberInput` component for the year. 

I removed tests in `DateInput` because those tests are now under the purview of `ErrorableNumberInput` and `ErrorableSelect`. Also had to add some specific styles for a date field with 2 dropdowns, and made 2 other minor typo fixes as well.

akainic committed:
Refactor DateInput by using ErrorableSelect and creating a new ErrorableNumberInput component

crwallace commented on _health-care/_js/utils/options-for-select.js:
```diff
@@ -329,6 +329,36 @@ const suffixes = [
+  11: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30'],
+  12: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31']
+};
+
```
I would be interested to see if there is a better way to write this so it isn't so redundant.

akainic replied:
Yeah... I was thinking about that. I was thinking of defining 3 variables: 
```javascript
const 31DayMonths = ['1', '2', ... '31']
const 30DayMonths = ['1', '2', ... '30']
const 29DayMonths = ['1', '2', ... '29']
```
And then within `days` just pointing to those instead?

crwallace replied:
Yeah that seems like a good structure. This might be a TODO to address later.

crwallace commented on _health-care/_js/components/form-elements/ErrorableSelect.jsx:
```diff
@@ -63,7 +63,7 @@ class ErrorableSelect extends React.Component {
         <label
-            className={this.props.errorMessage ? 'usa-input-error-label' : undefined}
+            className={this.props.errorMessage !== undefined ? 'usa-input-error-label' : undefined}
             htmlFor={this.selectId}>
```
Will this make the field look wrong until the user types something in because it will be `undefined` on initialization correct?

akainic replied:
Yeah, this isn't very clear. I added the check for `undefined` because the way we add the error CSS classes is based on the presence of `errorMessage`, which isn't always necessarily correlated. For example, for DateInput I want the error CSS classes to be applied if there is an error but I don't need an error message to be displayed, so I set `errorMessage` to be an empty string, so no actual message appears, but it won't be undefined, so the proper class can be applied. I was thinking of changing the entire structure to not be dependent on `errorMessage`, since it seems like an imperfect correlation anyway, but I didn't want to tackle that in this PR. I can add a TODO to address later.

crwallace replied:
Ok sounds good!

crwallace commented on _health-care/_js/components/form-elements/ErrorableNumberInput.jsx:
```diff
@@ -0,0 +1,90 @@
+    return (
+      <div className={this.props.errorMessage ? 'usa-input-error' : undefined}>
+        <label
+            className={this.props.errorMessage !== undefined ? 'usa-input-error-label' : undefined}
```
Same as another comment, "Will this make the field look wrong until the user types something in because it will be undefined on initialization correct?"

crwallace commented:
LGTM with TODOs!

akainic commented:
@crwallace Thanks for the review! I added a TODO for the errorMessage issue and I refactored the days options like we talked about. Going to rebase and squash, and once the build passes I'll merge!
